src/libs/kata-types/src/config/hypervisor/dragonball.rs

```diff
@@ -65,6 +65,10 @@ impl ConfigPlugin for DragonballConfig {
                 db.cpu_info.default_maxvcpus = default::MAX_DRAGONBALL_VCPUS;
             }
 
+            if db.cpu_info.default_vcpus as u32 > db.cpu_info.default_maxvcpus {
+                db.cpu_info.default_vcpus = db.cpu_info.default_maxvcpus as i32;
```

Comment:
Any reason we are not defining `default_vcpus` as `u32`?

Reply:
1. When `default_vcpus` < 0, will be set to the actual number of physical cores.
2. When 0 < `default_vcpus` <= number of physical cores, will be set to the specified number

Reply:
Then what about changing the first condition to:

when `default_vcpus` == 0, set the vcpu number to the actual number of physical cores.

Then `default_vcpus` can be `u32`.

Reply:
@bergwolf I just checked our design. There are three cases for `default_vcpus`.

1. when `default_vcpus unspecified or 0. it will be set to `default::DEFAULT_GUEST_VCPUS`
2. when `default_vcpus < 0, will be set to the actual number of physical cores.
3. when 0 < `default_vcpus` <= number of physical cores, will be set to the specified number

So, `default_vcpus` has to be `i32`

Reply:
@Tim-0731-Hzt I see, this is inherited from go runtime. We can optimize but let's not block the PR here. Thanks for explanation.